airflow/models/baseoperator.py

```diff
@@ -1119,6 +1121,12 @@ def set_xcomargs_dependencies(self) -> None:
                 arg = getattr(self, field)
                 XComArg.apply_upstream_relationship(self, arg)
 
+    def iter_xcom_dependencies(self) -> Iterator[tuple[Operator, str]]:
+        """Upstream dependencies that provide XComs used by this operator."""
+        from airflow.models.xcom_arg import XComArg
+
+        yield from XComArg.iter_xcom_references(self)
```

Comment:
Why is this needed?

Reply:
Is it a question why we need to define a method for that instead of calling `XComArg.iter_xcom_references` directly? As I see that, when we look at a `task` and want to understand, which XComs it depends on, `XComArg.iter_xcom_references(task)` seems to me like a lower level detail of implementation. Something similar happens in the mapped operator:

https://github.com/apache/airflow/blob/3f6b5574c61ef9765d077bdd08ccdaba14013e4a/airflow/models/mappedoperator.py#L638-L643

Reply:
Is the recommendation to call `XComArg.iter_xcom_references` directly then, right?